[R25.09] feat: added first version of the data trust & security KIT (#1298)
eclipse-tractusx/eclipse-tractusx.github.io: docs-kits/kits/data-trust-and-security-kit/software-development-view/semantic-verification.md

    ],
    "type": [
        "VerifiableCredential",
        "DataAttestationCredential",
        "Pcf"
    ],
    "credentialSubject": {
        "@type": "urn:samm:io.catenax.pcf:7.0.0#Pcf",
        "specVersion": "urn:io.catenax.pcf:datamodel:version:7.0.0",
        "companyIds": ["https://example.com/company1"],
        "pcf": {
            "pcfExcludingBiogenic": 2.5,
            "fossilGhgEmissions": 1.8
        }
    },
    "issuer": "did:web:company.example.com",
    "validFrom": "2024-01-15T10:30:00Z",
    "validUntil": "2025-01-15T10:30:00Z",
    "proof": {
        "type": "Ed25519Signature2020",
        "created": "2024-01-15T10:30:00Z",
        "verificationMethod": "did:web:company.example.com#key-1",
        "proofPurpose": "assertionMethod",
        "proofValue": "z5vDVKmhQKyiPj..."
    }
}
```

## Semantic Validation Process

### Validation Workflow

The semantic validation process follows these steps:

```mermaid
sequenceDiagram
    participant Workflow as Workflow
    participant Parser as sammSchemaParser
    participant SemanticHub
    participant VC as Verifiable Credential

>     participant IssuanceService as Issance Service

Reviewer: Issance --> Issuance

---
id: semantic-verification
title: Semantic Context Verification
description: 'Using JSON-LD to verify SAMM modelled data models'
---

![Data Trust & Security KIT Icon](@site/static/img/kits/data-trust-and-security/data-trust-and-security-kit-logo.svg)

## Enabling SAMM Standardized Aspect Models with Verifiable Credentials

In order to enable the Eclipse Tractus-X Semantic Models to be used in Verifiable Credentials there is the need to generate JSON-LD "@context" for the data format.

Therefore, for making these two technologies compatible, it is necessary to make a translation from the standard JSON Schemas to the JSON-LD "@context" for the desired aspect models.

## Creating the @context

The solution is to create automatic translation between these standards:

This diagram shows the current incompatibility:

```mermaid

> flowchart TD

Author: ```suggestion
flowchart TD
      VC["Verifiable Credentials"] -- USE --> JLD["JSON-LD"]
      JLD["JSON-LD"] -- REQUIRES --> JLDC["@context"]
      SAMM["SAMM"] -- GENERATES NO --> JLDC["@context"]
      SM["TX Semantic Models"] -- ARE MODELLED IN --> SAMM["SAMM"]
      SAMM["SAMM"] -- GENERATES --> JSS["JSON SCHEMA"]
      JSS["JSON-SCHEMA"] -- CAN BE TRANSLATED TO --> JLDC["@context"]
      SAMM["SAMM"] -- CAN EXPORT SEMANTICS AS --> OWLRDF["OWL/RDF"]
      OWLRDF["OWL/RDF"] -- GENERATE --> JLDC["@context"]
```

    VC["Verifiable Credentials"] -- USE --> JLD["JSON-LD"]
    JLD["JSON-LD"]  -- REQUIRES --> JLDC["@context"]
    SAMM["SAMM"] -- GENERATES NO --> JLDC["@context"]
    SM["TX Semantic Models"] -- ARE MODELLED IN --> SAMM["SAMM"]
    SAMM["SAMM"] -- GENERATES --> JSS["JSON SCHEMA"]
    JSS["JSON-SCHEMA"] -- CAN BE TRANSLATED TO --> JLDC["@context"]
```

The reality is:

```mermaid
flowchart LR
    SM["Tractus-X Data Models"] x--NOT COMPATIBLE WITHOUT TRANSLATION --x VC["Verifiable Credentials"]
```

> 

Author: ```suggestion
So some sort of translation is necessary. Or directly from the JSON-Schemas or using the OWL or RDF format.

```

### The Translation Solution

> 

Author: we can include that in the next kIT release.

> 

Author: I have added a disclaimer.

> We solve this by automatically converting SAMM JSON schemas into JSON-LD contexts:

Author: ```suggestion
:::info

More information about translating SAMM into OWL and RDF and then translating to JSON-LD context is still not available in this KIT. It will be available probably in the next version.

:::

One way of solving this is by automatically converting SAMM JSON schemas into JSON-LD contexts:

```
